Tests/Marketplace/zip_packs.py

```diff
@@ -114,61 +114,84 @@ def remove_test_playbooks_from_signatures(path, filenames):
+        if entry not in IGNORED_FILES and entry in packs_list and os.path.isdir(entry_path):
+            # This is a pack directory, should keep only most recent release zip
+            pack_files = get_pack_files(entry_path)
+            latest_zip = get_latest_pack_zip_from_blob(entry, pack_files)
+            if not latest_zip:
+                logging.warning(f'Failed to get the zip of the pack {entry} from GCP')
                 continue
-
-            if gcp_path == BUILD_GCP_PATH:
-                pack_prefix = os.path.join(gcp_path, branch_name, circle_build, 'content', 'packs', pack.name)
-            else:
-                pack_prefix = os.path.join(gcp_path, branch_name, circle_build, pack.name)
-
-            if not branch_name or not circle_build:
-                pack_prefix = pack_prefix.replace('/builds/content', '')
-
-            # Search for the pack in the bucket
-            blobs = list(storage_bucket.list_blobs(prefix=pack_prefix))
-            if blobs:
-                blob = get_latest_pack_zip_from_blob(pack.name, blobs)
-                if not blob:
-                    logging.warning(f'Failed to get the zip of the pack {pack.name} from GCP')
-                    continue
-                download_path = os.path.join(destination_path, f"{pack.name}.zip")
-                zipped_packs.append({pack.name: download_path})
-                logging.info(f'Downloading pack from GCP: {pack.name}')
-                executor_submit(executor, download_path, blob)
-                sleep(1)
-                if os.path.exists('/home/runner/work/content-private/content-private/content/artifacts/'):
-                    logging.info(f"Copying pack from {download_path} to /home/runner/work/content-private/"
-                                 f"content-private/content/artifacts/packs/{pack.name}.zip")
-                    shutil.copy(download_path,
-                                f'/home/runner/work/content-private/content-private/content/artifacts/'
-                                f'packs/{pack.name}.zip')
-            else:
-                logging.warning(f'Did not find a pack to download with the prefix: {pack_prefix}')
+            logging.info(f"Found latest zip of {entry}, which is {latest_zip}")
+            zipped_packs.append({Path(latest_zip).stem: latest_zip})
 
     if not zipped_packs:
-        logging.critical('Did not find any pack to download from GCP.')
-        sys.exit(1)
+        raise Exception('No zip files were found')
     return zipped_packs
 
 
+def download_packs_from_gcp(storage_bucket_name, dest_path):
+    """
+    Copies all content in the gcp path to destination_path
+    Args:
+        storage_bucket_name: The name of the GCP bucket to download from.
+        dest_path: The path to download the packs to.
+    """
+    src_path = os.path.join('gs://', storage_bucket_name, GCPConfig.STORAGE_BASE_PATH)
+
+    process = subprocess.check_output(['gsutil', '-m', 'cp', '-r', src_path, dest_path])
```

Comment:
Unless the service account has already been used to authenticate to Cloud Storage in a previous step in the build process... this won't work. We shouldn't make that assumption. We should leave passing the service account as a script parameter and perform authentication in the event that it is necessary.

Reply:
I'm was not aware of the service account. 
However, if it has already been authenticated in a previous step why should we authenticate again? What is the scenario where we'll need to re-authenticate? Is this step will ever be executed independently?